kube-derive/src/resource_inherit.rs

```diff
@@ -0,0 +1,188 @@
+// Generated by darling macros, out of our control
+#![allow(clippy::manual_unwrap_or_default)]
+
+use darling::{FromDeriveInput, FromMeta};
+use syn::{parse_quote, Data, DeriveInput, Path};
+
+/// Values we can parse from #[kube(attrs)]
+#[derive(Debug, FromDeriveInput)]
+#[darling(attributes(inherit))]
+struct InheritAttrs {
+    resource: syn::Path,
+    #[darling(default)]
+    namespaced: bool,
+    #[darling(default)]
+    dynamic: bool,
```

Comment:
why do you want to inherit dynamic scoped resources? dynamic is an escape hatch for indeterminate scopes

Reply:
I don’t think it is needed, just ended up exploring `DynamicObject` compatibility and went on a route of matching the expected type. This way that resource can be used with dynamic api, but likely it is unnecessary, after fixing https://github.com/kube-rs/kube/pull/1565#discussion_r1743455813

Reply:
yeah, my gut feeling is that we leave it out initially. people would generally get `ApiResource` as part of discovery and by then it's too late to derive.